### PR TITLE
Allow embedder to specify a vsync waiter.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/28240

### CODE REVIEW

File: shell/platform/android/vsync_waiter_android.cc [hunk 1]
```diff
@@ -43,11 +38,28 @@ void VsyncWaiterAndroid::AwaitVSync() {
...
+    return kUnknownRefreshRateFPS;
+  }
+  jclass clazz = g_vsync_waiter_class->obj();
+  if (clazz == nullptr) {
+    return kUnknownRefreshRateFPS;
+  }
+  jfieldID fid = env->GetStaticFieldID(clazz, "refreshRateFPS", "F");
+  // We can safely read this 32-bit float from Java in any thread because
+  // 32-bits read and write are guaranteed to be atomic:
+  // https://stackoverflow.com/questions/11459543/should-getters-and-setters-be-synchronized/11459616#11459616
```
Comment (jason-simmons) [inline shell/platform/android/vsync_waiter_android.cc]: I'd remove this URL.  Atomicity of float reads is defined in the Java language spec.

**Comment by cbracken** [comment]
Please hold off on landing until I've got #7913 landed (should be soon).

**Comment by chinmaygarde** [comment]
I believe Daniel (I don't know his GitHub handle) can patch this in locally to unblock himself as he prototypes the patch. Just wanted to post this up ASAP.

File: shell/platform/android/vsync_waiter_android.cc (hunk 1 above)
Comment (chinmaygarde) [reply shell/platform/android/vsync_waiter_android.cc]: I didn't add that link. Just moved that code around.

Comment (chinmaygarde) [reply shell/platform/android/vsync_waiter_android.cc]: I'll get rid of it anyway.

**Comment by chinmaygarde** [comment]
@cbracken: Is this good to go?

**Comment by cbracken** [comment]
Yep good to land
Looks like you'll need to resolve the merge conflict with jason's patch.

**Pull request merged**